keep/api/routes/providers.py

```diff
@@ -211,7 +211,12 @@ def delete_provider(
     )
     secret_manager = SecretManagerFactory.get_secret_manager()
     secret_name = f"{tenant_id}_{provider_type}_{provider_id}"
-    secret_manager.delete_secret(secret_name)
+    try:
+        secret_manager.delete_secret(secret_name)
+    except Exception as exc:
+        # TODO: handle it better
+        logger.exception("Failed to delete the provider secret")
+        pass
```

Comment:
This will show in the UI as if the provider was deleted but won't actually deleted (back after refresh)